### PR TITLE
FINERACT-2011: Savings Account - transaction order is incorrectly mix…

### PR DESCRIPTION
**Pull request by marta-jankovics**
…ed after certain activities

## Description

Describe the changes made and why they were made.

Ignore if these details are present on the associated [Apache Fineract JIRA ticket](https://github.com/apache/fineract/pull/1284).

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [ ] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [ ] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [ ] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [ ] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/domain/SavingsAccount.java [hunk 1]
```diff
@@ -811,108 +815,102 @@ public List<PostingPeriod> calculateInterestUsing(final MathContext mc, final Lo
...
-        // 2. determine latest 'posting period' and find interest credited to
-        // that period
-
-        // A generate list of EndOfDayBalances (not including interest postings)
-        final SavingsPostingInterestPeriodType postingPeriodType = SavingsPostingInterestPeriodType.fromInt(this.interestPostingPeriodType);
-
-        final SavingsCompoundingInterestPeriodType compoundingPeriodType = SavingsCompoundingInterestPeriodType
-                .fromInt(this.interestCompoundingPeriodType);
+        final List<PostingPeriod> allPostingPeriods = new ArrayList<>();
+        if (hasInterestCalculation() || hasOverdraftInterestCalculation()) {
```
Comment (adamsaghy) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/domain/SavingsAccount.java:827]: "hasOverDraftInterestCalculation" method is kind of unnecessary hence, if the "hasInterestCalculation()" is false, the "hasOverDraftInteretCalculation" method is false too (see last condition).

**Comment by adamsaghy** [comment]
@marta-jankovics Can you please help me understand where is the logic which makes the transaction order to be correct? I am not sure, i see it :/

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/domain/SavingsAccount.java (hunk 1 above)
Comment (marta-jankovics) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/domain/SavingsAccount.java:827]: hasInterestCalculation is checking the nominalAnnualInterestRate, while hasOverDraftInteretCalculation is checking the nominalAnnualInterestRateOverdraft

**Review by adamsaghy — APPROVED** [review]
LGTM

**Review by adamsaghy — APPROVED** [review]
LGTM

**Pull request merged**